### PR TITLE
Feat!: Migrate to ink! v4.0.0

### PR DESCRIPTION
**Pull request by SHA888**
- Breaking changes. 
- For more details, follow this [docs](https://use.ink/faq/migrating-from-ink-3-to-4)
- Add identity verification flow (WIP)

### CODE REVIEW

File: lib.rs [hunk 1]
```diff
@@ -195,6 +183,51 @@ mod chocolate {
...
+                {
+                    return Err(Error::VerificationFlowAlreadyInitiated);
+                }
+            // Increment verification flow for uniqueness
+            self.verifications_count = self.verifications_count.saturating_add(1);
+
+            // Combine account and id for unique signable message
+            let mut verification_message = self.env().caller().encode(&mut Vec::new());
+            todo!("Add verifications_count to verification message");
+            Ok(todo!("Sign verification message"))
```
Comment (islami00) [inline lib.rs]: This part has been filled here https://pastebin.com/8YTPj7tH,
 Is there a reason why it's todo?

File: lib.rs [hunk 2]
```diff
@@ -22,62 +18,41 @@ mod chocolate {
         reviews: Mapping<u32, Review>,
         /// Accountid + projectId
         reviews_projects_list: Vec<(AccountId, u32)>,
+        /// Stores the address of account that have initiated the verification flow.
+        account_verification_flow_initiation: Mapping<AccountId, bool>, // TODO: Maybe use a bool?
```
Comment (islami00) [inline lib.rs]: This should map to the message we send out, then we can verify that it's the right signature we got back

**Review by islami00 — CHANGES_REQUESTED** [review]
LGTM overall,
I would also recommend checking with the steps in [chocolate-wasm](https://github.com/chocolatenetwork/chocolate-wasm/blob/main/STEPS.md), and also if [typegen](https://www.npmjs.com/package/@astar-network/swanky-cli#swanky-contract-typegen-contractname) still works with swanky, as that would affect the frontend

File: lib.rs (hunk 1 above)
Comment (SHA888) [reply lib.rs]: Yeah, some part of the code didn't work, so this is work in progress (WIP)

File: lib.rs (hunk 2 above)
Comment (SHA888) [reply lib.rs]: Okay, the storage for AccountId in ink 4.0 has been changed. I'm still figuring it out. Thanks for the heads up.

**Comment by SHA888** [comment]
Resolved in #7

**Pull request closed without merging**